src/fs/metadata_log.cc

```diff
@@ -100,18 +100,65 @@ void metadata_log::write_update(inode_info::file& file, inode_data_vec new_data_
     } else {
         cut_out_data_range(file, new_data_vec.data_range);
     }
-
     file.data.emplace(new_data_vec.data_range.beg, std::move(new_data_vec));
 }
 
 void metadata_log::cut_out_data_range(inode_info::file& file, file_range range) {
-    file.cut_out_data_range(range, [](inode_data_vec data_vec) {
-        (void)data_vec; // TODO: for compaction: update used inode_data_vec
+    file.cut_out_data_range(range, [&](const inode_data_vec& former, const inode_data_vec& left_remains, const inode_data_vec& right_remains) {
+        if (left_remains.data_range.size() > 0) {
+            std::visit(overloaded {
+                [&](const inode_data_vec::in_mem_data&) {
+                    _rewrite_log_size += ondisk_entry_size<ondisk_small_write_header>(left_remains.data_range.size());
+                },
+                [&](const inode_data_vec::on_disk_data& disk_data) {
+                    // Remaining data is smaller than original so it is not a large write anymore
+                    _rewrite_log_size += ondisk_entry_size<ondisk_medium_write>();
+                },
+                [&](const inode_data_vec::hole_data&) {
+                },
+            }, left_remains.data_location);
+        }
+        if (right_remains.data_range.size() > 0) {
+            std::visit(overloaded {
+                [&](const inode_data_vec::in_mem_data&) {
+                    _rewrite_log_size += ondisk_entry_size<ondisk_small_write_header>(right_remains.data_range.size());
+                },
+                [&](const inode_data_vec::on_disk_data& disk_data) {
+                    _rewrite_log_size += ondisk_entry_size<ondisk_medium_write>();
+                },
+                [&](const inode_data_vec::hole_data&) {
+                },
+            }, right_remains.data_location);
+        }
+        std::visit(overloaded {
+            [&](const inode_data_vec::in_mem_data&) {
+                size_t log_decrease = ondisk_entry_size<ondisk_small_write_header>(former.data_range.size());
+                _rewrite_log_size -= log_decrease;
+            },
+            [&](const inode_data_vec::on_disk_data& disk_data) {
+                // TODO: Differentiate between large write and large write without mtime
+                size_t log_decrease = (former.data_range.size() == _cluster_size ?
+                    ondisk_entry_size<ondisk_large_write>() :
+                    ondisk_entry_size<ondisk_medium_write>());
+                _rewrite_log_size -= log_decrease;
+                // Remaining data is smaller than original so it is not a large write anymore
+                _data_cluster_contents_info_map[offset_to_cluster_id(disk_data.device_offset, _cluster_size)]->cut_data(
```

Comment:
```suggestion
                _data_cluster_contents_info_map.at(offset_to_cluster_id(disk_data.device_offset, _cluster_size))->cut_data(
```
Use at(), because operator[] creates element if it does not exist and that is not what we want here -- for us it is a bug. Although an assert is preferable for this, at() throws exception and there is no version of at() that would assert that the key exists.

Reply:
```suggestion
                _data_cluster_contents_info_map.find(offset_to_cluster_id(disk_data.device_offset, _cluster_size))->second->cut_data(
```
That is the second option that I would like better, but there is no checking here so it is better to write it in that way:
```suggestion
                auto it = _data_cluster_contents_info_map.find(offset_to_cluster_id(disk_data.device_offset, _cluster_size));
                assert(it != _data_cluster_contents_info_map.end());
                it->second->cut_data(
```
But the most preferable would be some version of at that asserts instead of throws, eg.
```suggestion
                our_at(_data_cluster_contents_info_map, offset_to_cluster_id(disk_data.device_offset, _cluster_size))->cut_data(
```